backend/main-service/src/main/java/kkakka/mainservice/product/ui/ProductController.java

```diff
@@ -43,15 +47,23 @@ public ResponseEntity<ProductDetailResponse> showProductDetail(
     }
 
     @GetMapping
-    public ResponseEntity<PageableResponse<List<ProductResponse>>> showAllProducts(
-            @RequestParam(value = "category", required = false) Long categoryId,
-            @RequestParam(value = "sortBy", defaultValue = "NON", required = false) String sortBy,
-            @RequestParam(value = "keyword", required = false) String keyword,
-            Pageable pageable) {
-        final Page<ProductDto> productDtos = productService.showAllProductsWithCategoryAndSearch(
-                Optional.ofNullable(categoryId), sortBy, keyword, pageable);
-        return ResponseEntity.status(HttpStatus.OK)
-                .body(dtoToPageableResponse(pageable, productDtos));
+    public ResponseEntity<SearchResultResponse> showAllProducts(
+        @ModelAttribute SearchParamRequest searchParamRequest,
+        Pageable pageable) {
+        SearchResultResponse result;
+        if (searchParamRequest.isKeyword()) {
```

Comment:
```suggestion
        if (Objects.isNull(searchParamRequest.getKeyword()) {
```
dto 에서 판단하는거보다 이렇게 쓰면 어떨까요